server/sonar-web/src/main/js/apps/projects/store/utils.js

```diff
@@ -33,16 +34,64 @@ const getAsLevel = value => {
 
 export const parseUrlQuery = urlQuery => ({
   'gate': getAsLevel(urlQuery['gate']),
+  'reliability': getAsNumericRating(urlQuery['reliability']),
+  'security': getAsNumericRating(urlQuery['security']),
+  'maintainability': getAsNumericRating(urlQuery['maintainability']),
+  'coverage': getAsNumericRating(urlQuery['coverage']),
+  'duplications': getAsNumericRating(urlQuery['duplications']),
+  'size': getAsNumericRating(urlQuery['size']),
+});
 
-  'coverage__gte': getAsNumber(urlQuery['coverage__gte']),
-  'coverage__lt': getAsNumber(urlQuery['coverage__lt']),
+const convertCoverage = coverage => {
+  switch (coverage) {
+    case 1:
+      return 'coverage < 30';
+    case 2:
+      return 'coverage >= 30 and coverage < 50';
+    case 3:
+      return 'coverage >= 50 and coverage < 70';
+    case 4:
+      return 'coverage >= 70 and coverage < 80';
+    case 5:
+      return 'coverage >= 80';
+    default:
+      return '';
+  }
+};
 
-  'duplications__gte': getAsNumber(urlQuery['duplications__gte']),
-  'duplications__lt': getAsNumber(urlQuery['duplications__lt']),
+const convertDuplications = duplications => {
```

Comment:
![CRITICAL](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-critical.png) Function has a complexity of 12 which is greater than 10 authorized. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://dory.sonarsource.com/sonarqube/coding_rules#rule_key=javascript%3AFunctionComplexity)